samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,202 @@
+
+The downside of the focus on the communication patterns is that the saga design is less than ideal for a real voting system. There will be quite some contention on the saga data, which may result in concurrency exceptions and a few retries impacting performance of the system.
+
+For logging purposes a simple static logger was used to expose specific log statements for the routing part. It is recommended to use a dedicated logging package for production instead.
+
+## Solution structure
+
+The solution contains the following projects:
+
+ * **Contracts**: Contains message definitions that are shared among projects.
+ * **Shared**: Contains the receiver side distribution and the sender side distribution code.
+ * **Voter**: This is a console application that simulates casting of votes. It is hosted outside of the Service Fabric cluster. 
+ * **CandidateVoteCount**: Service Fabric service with the logic to count votes by candidate while the votes come in. It also instructs the `ZipCodeVoteCount` endpoint to track votes by zip code. It will report the intermediate results as well as the final results when the election is closed.
+ * **ZipCodeVoteCount**: Service Fabric service with the logic to count the votes by zip code in the background. It will report the results when the allowed counting period is over.
+ * **ServiceFabricRouting**: [Service Fabric application](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-application-model) containing description of services it will run when deployed into Service Fabric cluster.
+
+
+## Cluster partitioning
+
+The `CandidateVoteCount` is a [stateful service](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-introduction) that uses a `NamedPartition` [partitioning scheme](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-concepts-partitioning). Each candidate has its own partition, so there is one called "John" and another called "Abby". 
+
+The `ZipCodeVoteCount` is a stateful service that uses a `UniformInt64Partition` partitioning scheme with the `PartitionCount` set to 3, the `LowKey` set to 0 and the `HighKey` set to 99000. This configuration ensure that the partition is split into 3 well-known ranges (0 -> 32999), (33000 -> 65999) , (66000 -> 98999).
+
+
+## Routing
+
+The default NServiceBus routing approach cannot be used as-is with Service Fabric stateful services. Stateful services assume business data partitioning. A message must be delivered to a specific replica (instance) of a stateful that can handle the message data. E.g. for each `PlaceVote` message associated with a cast vote, the message should be routed to the partition associated with the preferred candidate, "John" or "Abby". Similar to that, to count the votes per zip code the `TrackZipCode` message needs to delivered to the partition that is responsible for the range to which the zip code belongs.
+
+Example:
```

Comment:
I would skip the details here. As noted in https://github.com/Particular/docs.particular.net/pull/2435#discussion_r103675750 the implementation of the hashing function is a business level detail that is probably not the most important here.